### ISSUE TITLE
Change Password Error: postgres is a system account and cannot be used

### ISSUE THREAD

**Issue by srinathganesh** [CrunchyData/postgres-operator#2169]
** Which example are you working with? **
Change password for default postgres user

**What is the current behavior?**
Error: postgres is a system account and cannot be used

**What is the expected behavior?**

**Other information** (e.g. detailed explanation, related issues,  etc)

**Please tell us about your environment:**
```
$ pgo version
pgo client version 4.4.1
pgo-apiserver version 4.4.1
```

* Operating System: Crunchydata Kubernetes Helm Chart DEFAULTS
* Where is this running ( Local , Cloud Provider): Kubernetes Cluster
* Storage being used (NFS, Hostpath, Gluster, etc): Any Kubernetes PVC
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* crunchy-postgresql-operator Helm chart `postgres-operator` 
* home: https://github.com/CrunchyData/postgres-operator
* version: 0.1.0
* appVersion: 4.4.1

Scipt to change user password
```
kubectl exec -it pgo-client-7f5849c-7zhwf -- bin/bash
pgo update user --selector=mycluster --username=postgres --password=newpass 
```

**Error: postgres is a system account and cannot be used**

**Comment by jkatz** [comment]
You can retrieve the password with the `--show-system-accounts1` flag:

```
pgo show user --show-system-accounts mycluster
```

**Comment by srinathganesh** [comment]
I don't want to retrieve it. I just wanted to change it. I do have the
current password (correct one)

**Comment by srinathganesh** [comment]
Hi, to just rephrase the problem (if more clarity is needed)

- I have admin password (existing)
- I want to change the admin password

Let me know if you need more details

**Comment by jkatz** [comment]
I understand the case. The `superuser` is technically a managed account and I would need to evaluate the impact on allowing for that password to be changed via the Operator. That said, the Operator's operations which require interfacing as `postgres` use a different authentication method other than a password based one.

You can change this password directly in the PostgreSQL cluster using `ALTER ROLE` or, even better, the `\password` command. You can do so by directly accessing the PostgreSQL instance from the Pod.

**Comment by srinathganesh** [comment]
ok thank you. I did real about postgres internal password change but my
fear was it should not have impact on Scaling/Clustering so I didn't
proceed with that.

Will give it a try and keep you posted.

**Comment by srinathganesh** [comment]
**Update**:

- Postgres change password works great (even with replicas on) 
- Also any application connecting with the Database only required Postgres password (and not of Crunchydata)

---------------------

Given enhanced security requirements, is there a possibility that the *old* password of the Operator be misused to mess around? This problem alleviated if there was a way to change it.

**Comment by jkatz** [comment]
> Given enhanced security requirements, is there a possibility that the old password of the Operator be misused to mess around?

No. Once the password is changed in Postgres, the password that is contained in the Secret maintained by the Operator is invalid (unless you change the password in Postgres to match the password in the Secret). Postgres does not maintain a password history at this time.

**Comment by jkatz** [comment]
#2110 adds support for setting these passwords -- you will have to specify the `--set-system-account-password` flag with `pgo update user` -- and it will be available in 4.6.0.
